source/brailleDisplayDrivers/albatross/driver.py

```diff
@@ -0,0 +1,758 @@
+		# Also checking that _currentCells has sometimes updated.
+		# Show last known content.
+		if (
+			len(self._oldCells) == self.numCells
+			and len(self._oldCells) == len(self._currentCells)
+		):
+			self._clearOldCells()
+			self.display(self._currentCells)
+			log.debug(
+				"Updating display content after reconnection or display menu exit"
+			)
+		if self._waitingSettingsByte:
+			self._waitingSettingsByte = False
+
+	def _handleSettingsByte(self, data: bytes):
+		"""Extract current settings from settings byte.
+		All other settings except number of cells are only notes to screenreader,
+		and it is screenreader or driver job to use them when applicable.
+		For example, there are no separate status cells in the device but if
+		screenreader supports using status cells, it can be notified to use them
+		by settings byte.
+
+		Settings byte contain following settings (bits referred by using
+		MSB 0 scheme):
+
+		- bit 0: number of cells; 0 = 46, 1 = 80. This is model based value
+		which cannot be changed. This is the most important setting, and it
+		must be applied.
+
+		- bit 1: switch left side and right side keys; 0 = no, 1 = yes.
+		Left, right, down3, up2, routing and secondRouting keys are not affected.
+		Up2 and down3 are ignored because they are in the middle of the front
+		panel of 80 model so they do not logically belong to left or right side.
+		All other keys are switched with corresponding other side keys.
+
+		- bit 2: place of status cells; 0 = left, 1 = right. Not implemented.
+		NVDA does not use status cells at the moment.
+
+		- bit 3: all keys act as right side keys; 0 = no, 1 = yes.
+		Left, right, down3, up2, routing and secondRouting keys are not affected.
+		Up2 and down3 are ignored because they are in the middle of the front
+		panel of 80 model so they do not logically belong to left or right side.
+		All other left side keys are assigned to corresponding right side keys.
+
+		- bit 1 = 0 and bit 3 = 0: normal key layout.
+
+		- bit 1 = 1 and bit 3 = 1: all keys act as corresponding left side keys.
+		Left, right, down3, up2, routing and secondRouting keys are not affected.
+		Up2 and down3 are ignored because they are in the middle of the front
+		panel of 80 model so they do not logically belong to left or right side.
+		All other right side keys are assigned to corresponding left side keys.
+
+		- bits 4 - 7: number of status cells. Not implemented.
+		NVDA does not use status cells at the moment.
+
+		@param data: Settings byte
+		"""
+		self.numCells = 80 if ord(data) >> 7 == 1 else 46
+		self._keyLayout = ord(data) >> 4 & KEY_LAYOUT_MASK
+		log.debug(
+			f"Current settings: number of cells {self.numCells}, "
+			f"key layout {KeyLayout(self._keyLayout).name}")
+
+	def _handleKeyPresses(self, data: bytes):
+		"""Handles display button presses.
+		in single ctrl-key presses and ctrl-key combinations the first key is
+		resent as last one.

```

Comment:
I don't quite understand this. Is this suggesting the ctrl key is always last?

```suggestion
		In single ctrl-key presses and ctrl-key combinations the first key is
		present as last one.
```